src/Tokenizer.zig

```diff
@@ -1453,7 +1444,7 @@ pub fn next(self: *Tokenizer) Token {
         switch (state) {
             .start, .line_comment => {},
             .u, .u8, .U, .L, .identifier, .extended_identifier => {
-                id = Token.getTokenId(self.comp, self.buf[start..self.index], state.identifierType());
+                id = if (state != .extended_identifier) Token.getTokenId(self.comp, self.buf[start..self.index]) else .extended_identifier;
```

Comment:
It'd be easier to just add a `.extended_identifier => id = .extended_identifier,` prong.

Same could be said for the previous case but at least that has some logic beyond just a single assignment.